flow/datatypes/numeric.go

```diff
@@ -1,12 +1,76 @@
 package datatypes
 
 const (
-	PeerDBNumericPrecision    = 38
-	PeerDBNumericScale        = 20
+	// defaults
+	PeerDBBigQueryPrecision   = 38
+	PeerDBBigQueryScale       = 20
+	PeerDBSnowflakePrecision  = 38
+	PeerDBSnowflakeScale      = 20
 	PeerDBClickhousePrecision = 76
 	PeerDBClickhouseScale     = 38
 )
 
+type WarehouseNumericComaptibility interface {
+	MaxPrecision() int16
+	MaxScale() int16
+	DefaultPrecisionAndScale() (int16, int16)
+	IsValidPrevisionAndScale(precision, scale int16) bool
+}
+
+type ClickHouseNumericCompatibility struct{}
+
+func (c ClickHouseNumericCompatibility) MaxPrecision() int16 {
+	return 76
+}
+
+func (c ClickHouseNumericCompatibility) MaxScale() int16 {
+	return 38
+}
+
+func (c ClickHouseNumericCompatibility) DefaultPrecisionAndScale() (int16, int16) {
+	return PeerDBClickhousePrecision, PeerDBClickhouseScale
+}
+
+func (c ClickHouseNumericCompatibility) IsValidPrevisionAndScale(precision, scale int16) bool {
+	return precision > 0 && precision <= PeerDBClickhousePrecision && scale < precision
+}
+
+type SnowflakeNumericCompatibility struct{}
+
+func (c SnowflakeNumericCompatibility) MaxPrecision() int16 {
+	return 38
+}
+
+func (c SnowflakeNumericCompatibility) MaxScale() int16 {
+	return 37
+}
+
+func (c SnowflakeNumericCompatibility) DefaultPrecisionAndScale() (int16, int16) {
+	return PeerDBSnowflakePrecision, PeerDBSnowflakeScale
+}
+
+func (c SnowflakeNumericCompatibility) IsValidPrevisionAndScale(precision, scale int16) bool {
+	return precision > 0 && precision <= 38 && scale < precision
+}
+
+type BigQueryNumericCompatibility struct{}
+
+func (c BigQueryNumericCompatibility) MaxPrecision() int16 {
```

Comment:
```suggestion
func (BigQueryNumericCompatibility) MaxPrecision() int16 {
```
can omit unused receivers